code/modules/robotics/bot/secbot.dm

```diff
@@ -193,8 +234,81 @@
 			if(radio_controller)
 				radio_controller.add_object(src, "[control_freq]")
 				radio_controller.add_object(src, "[beacon_freq]")
+			if (src.botgun == "taser")
+				add_loot = /obj/item/gun/energy/taser_gun
+				proj = new/datum/projectile/energy_bolt/robust
+				spread = 0
+				burst = 3
+				refire = 2
+				gun = "taser"
+				gunhat = "taserburst100"
+			else if (src.botgun == "tshotty")
+				add_loot = /obj/item/gun/energy/tasershotgun
+				proj = new/datum/projectile/energy_bolt/tasershotgun
+				spread = 3
+				burst = 6
+				refire = 0
+				gun = "taser shotgun"
+				gunhat = "tasers"
+			else if (src.botgun == "pulse")
+				add_loot = /obj/item/gun/energy/pulse_rifle
+				proj = new/datum/projectile/energy_bolt/pulse
+				spread = 0
+				burst = 2
+				refire = 5
+				gun = "pulse rifle"
+				gunhat = "pulse_rifle"
+			else if (src.botgun == "egun")
+				add_loot = /obj/item/gun/energy/egun
+				proj = new/datum/projectile/laser
+				spread = 1
+				burst = 2
+				refire = 5
+				gun = "energy gun"
+				gunhat = "energykill100"
+			else if (src.botgun == "phaser")
+				add_loot = /obj/item/gun/energy/phaser_gun
+				proj = new/datum/projectile/laser/light
+				spread = 1
+				burst = 3
+				refire = 5
+				gun = "phaser gun"
+				gunhat = "phaser-new"
+			else if (src.botgun == "laser")
+				add_loot = /obj/item/gun/energy/laser_gun
+				proj = new/datum/projectile/laser
+				spread = 1
+				burst = 1
+				refire = 5
+				gun = "laser gun"
+				gunhat = "laser"
+			else if (src.botgun == "ak")
+				add_loot = /obj/item/gun/kinetic/ak47
+				proj = new/datum/projectile/bullet/ak47
+				spread = 1
+				burst = 2
+				refire = 5
+				gun = "AK-744"
+				gunhat = "ak47"
+			else
+				add_loot = null 
+				proj = null
+				spread = 0
+				burst = 0
+				refire = 5
+				gun = null
```

Comment:
should be changed as per var changes above (put projectile modifiers on projectile, not var on bot, etc)

Reply:
Done! While it does limit some of the badmin-side flexibility (like 360 degree rapidfire BFGs), the new implementation that all yall gosh-darned legends taught me how to do is sooo much easier to work with and doesn't require a bunch of duplicate coding just to make it pretend to use a different gun. Thanks again!